File: ex_5/store/frontend/src/services/userAddress.ts
import axios from 'axios';

const HOST = 'http://localhost:9000'

export const addUserAddress = async (firstname: string, lastname: string, address: string, zipcode: string, city: string, country: string) => {
	return axios.post(`${HOST}/api/user-address/create`, {
		firstname: firstname,
		lastname: lastname,
		address: address,
		zipcode: zipcode,
		city: city,
		country: country,
	})
}

File: ex_6/store/backend/app/models/UserAddress.scala
package models

import models.util.TimestampFormatter
import play.api.libs.json._

import java.sql.Timestamp
import java.time.Instant

case class UserAddress(id: Long = 0, userId: Long, firstname: String, lastname: String, address: String,
                       zipcode: String, city: String, country: String, createdAt: Timestamp = Timestamp.from(Instant.now()), updatedAt: Timestamp = Timestamp.from(Instant.now()))

object UserAddress extends TimestampFormatter {
  implicit val userAddressFormat: OFormat[UserAddress] = Json.using[Json.WithDefaultValues].format[UserAddress]
}
